src/bundle/Resources/public/js/CKEditor/custom-tags/custom-tag-editing.js

```diff
@@ -0,0 +1,135 @@
+    static get requires() {
+        return [Widget];
+    }
+
+    defineSchema() {
+        const schema = this.editor.model.schema;
+
+        schema.register('customTag', {
+            isObject: true,
+            allowWhere: '$block',
+            allowAttributes: ['customTagName', 'values'],
+        });
+
+        schema.register('customTagContent', {
+            isBlock: true,
+            allowIn: 'customTag',
+            allowContentOf: '$root',
+        });
+    }
+
+    defineConverters() {
+        const conversion = this.editor.conversion;
+
+        conversion.for('editingDowncast').elementToElement({
+            model: 'customTag',
+            view: (modelElement, { writer: downcastWriter }) => {
+                const container = downcastWriter.createContainerElement('div', {
+                    'data-ezelement': 'eztemplate',
+                    'data-ezname': modelElement.getAttribute('customTagName'),
+                    class: 'ez-custom-tag',
+                });
+
+                return toWidget(container, downcastWriter);
+            },
+        });
+
+        conversion.for('dataDowncast').elementToElement({
+            model: 'customTag',
+            view: (modelElement, { writer: downcastWriter }) => {
+                const container = downcastWriter.createContainerElement('div', {
+                    'data-ezelement': 'eztemplate',
+                    'data-ezname': modelElement.getAttribute('customTagName'),
+                });
+                const config = downcastWriter.createUIElement('span', { 'data-ezelement': 'ezconfig' }, function(domDocument) {
+                    const domElement = this.toDomElement(domDocument);
+
+                    domElement.innerHTML = Object.entries(modelElement.getAttribute('values')).reduce((total, [attribute, value]) => {
+                        const ezvalue = `<span data-ezelement="ezvalue" data-ezvalue-key="${attribute}">${
+                            value !== null ? value : ''
+                        }</span>`;
+
+                        return `${total}${ezvalue}`;
+                    }, '');
+
+                    return domElement;
+                });
+
+                downcastWriter.remove(downcastWriter.createRangeIn(container));
+                downcastWriter.insert(downcastWriter.createPositionAt(container, 0), config);
+
+                return container;
+            },
+        });
+
+        conversion.for('upcast').elementToElement({
+            view: {
+                name: 'div',
+                attributes: {
+                    'data-ezelement': 'eztemplate',
+                },
+            },
+            model: (viewElement, { writer: upcastWriter }) => {
+                const configElement = viewElement.getChild(1);
+                const configValuesIterator = configElement.getChildren();
+                const customTagName = viewElement.getAttribute('data-ezname');
+                const values = {};
```

Comment:
```
const values = configValuesIterator.reduce((output, configValue) => ({
  ...output,
  [configValue.getAttribute('data-ezvalue-key')]: configValue.getChild(0) && configValue.getChild(0).data ? configValue.getChild(0).data : null
}), {});
```
maybe instead of whole for loop?